trace/obfuscate.go

```diff
@@ -22,7 +64,69 @@ type obfuscationVisitor struct {
 
 var _ vis.DefaultSpecVisitor = (*obfuscationVisitor)(nil)
 
+// EnterData processes the given data and obfuscates all the primitive values with zero values, regardless of it's meta data.
 func (*obfuscationVisitor) EnterData(self interface{}, ctx vis.SpecVisitorContext, d *pb.Data) Cont {
+	return ObfuscateWithZeroValue(d)
+}
+
+type partialObfuscationVisitor struct {
+	vis.DefaultSpecVisitorImpl
+	obfuscationOptions *Obfuscator
+}
+
+var _ vis.DefaultSpecVisitor = (*partialObfuscationVisitor)(nil)
+
+// EnterData processes the given data and obfuscates sensitive information based on the provided obfuscation options.
+// It does 3 checks to determine if the data contains sensitive information:
+// 1. It checks if spec is an HTTP Authroization or Cookie data and obfuscates the value.
+// 2. It cheks if spec is an HTTP Header or Query Param data and obfuscates the value if the key is in the list of sensitive keys.
+// 3. It applies regex patterns to all obfuscate sensitive primitive string values.
+func (pov *partialObfuscationVisitor) EnterData(self interface{}, ctx vis.SpecVisitorContext, d *pb.Data) Cont {
+	dataMeta, ok := d.GetMeta().GetMeta().(*pb.DataMeta_Http)
+	if ok {
+		var key string
+		switch dataMeta.Http.Location.(type) {
```

Comment:
There's a helper for extracting HTTP metadata directly.

```suggestion
	if httpMeta := d.GetMeta().GetHttp(); httpMeta != nil {
		var key string
		switch httpMeta.Location.(type) {
```